benchmarks/nonlinear_channel_flow/simple_nonlinear.cc

```diff
@@ -212,7 +215,10 @@ namespace aspect
                            * 0.5 * ((1/stress_exponent)-1) * std::pow(2,2) * out.viscosities[i] * (1/(edot_ii*edot_ii)) * deviator(in.strain_rate[i]),
                            * but we can (and do) simply it as:
                            */
-                          composition_viscosities_derivatives[c] = 2.0 * (stress_exponent_inv-1) * composition_viscosities[c] * (1.0/(edot_ii*edot_ii)) * deviator(in.strain_rate[i]);
+                          composition_viscosities_derivatives[c] = 2.0 * (stress_exponent_inv-1) * composition_viscosities[c] * (1.0/(edot_ii*edot_ii)) * deviator_strain_rate;
+
+                          if (use_deviator_of_strain_rate == true)
+                            composition_viscosities_derivatives[c]*deviator_tensor<dim>();
```

Comment:
This statement has no effect: you are computing the product of two things, but not assigning the result to anything. Can you explain what you wanted to do here?